cmd/go-yaml/event.go

```diff
@@ -491,11 +564,176 @@ func formatEventInfo(event *Event, profuse bool) *EventInfo {
 	}
 	if profuse {
 		if event.StartLine == event.EndLine && event.StartColumn == event.EndColumn {
-			info.Pos = fmt.Sprintf("%d;%d", event.StartLine, event.StartColumn)
+			// Single position
+			info.Pos = fmt.Sprintf("%d/%d", event.StartLine, event.StartColumn)
+		} else if event.StartLine == event.EndLine {
+			// Range on same line
+			info.Pos = fmt.Sprintf("%d/%d-%d", event.StartLine, event.StartColumn, event.EndColumn)
+		} else {
+			// Range across different lines
+			info.Pos = fmt.Sprintf("%d/%d-%d/%d", event.StartLine, event.StartColumn, event.EndLine, event.EndColumn)
```

Comment:
The classic way to write it would be this


```suggestion
			info.Pos = fmt.Sprintf("%d:%d", event.StartLine, event.StartColumn)
		} else if event.StartLine == event.EndLine {
			// Range on same line
			info.Pos = fmt.Sprintf("%d:%d-%d", event.StartLine, event.StartColumn, event.EndColumn)
		} else {
			// Range across different lines
			info.Pos = fmt.Sprintf("%d:%d-%d:%d", event.StartLine, event.StartColumn, event.EndLine, event.EndColumn)
```

Reply:
I made this change but it has the unfortunate side effect of quoting all the position values. That's why I chose the format that I did. Later though I think we can fix the encoder to not do that. It does that because of the YAML 1.1 rules. That's why I filed #231.